include/deal.II/fe/fe_hermite.h

```diff
@@ -0,0 +1,258 @@
+ * due to the increased regularity constraints many of these functions are
+ * shared between elements.
+ */
+template <int dim, int spacedim = dim>
+class FE_Hermite : public FE_Poly<dim, spacedim>
+{
+public:
+  /**
+   * Constructor that creates an Hermite finite element that imposes
+   * continuity in derivatives up to and including order @p regularity.
+   */
+  FE_Hermite<dim, spacedim>(const unsigned int regularity);
+
+  // Other functions
+  /**
+   * Returns <code>FE_Hermite<dim,spacedim>(regularity)</code> as a
+   * <code>std::string</code> with @p dim, @p spacedim and @p regularity
+   * replaced with the correct values.
+   */
+  virtual std::string
+  get_name() const override;
+
+  /**
+   * @copydoc dealii::FiniteElement::clone()
+   */
+  virtual std::unique_ptr<FiniteElement<dim, spacedim>>
+  clone() const override;
+
+  /**
+   * Returns the mapping between lexicographic and hierarchic numbering
+   * schemes for Hermite. See the class documentation for diagrams of
+   * examples of lexicographic numbering for Hermite elements.
+   */
+  std::vector<unsigned int>
+  get_lexicographic_to_hierarchic_numbering() const;
+
+  /**
+   * This re-implements FiniteElement::fill_fe_values() for a Hermite
+   * polynomial basis, to account for the more complicated shape function
+   * re-scaling that a Hermite basis requires. Since the idea of a Hermite
+   * basis is to strongly impose derivative continuities at the element
+   * boundaries, it is necessary to account for any changes to derivatives
+   * from those on the reference element due to the current cell mapping.
+   *
+   * At present this is done by only allowing certain cell mappings that
+   * guarantee rectangular cells which ensures that the directions of
+   * derivatives does not change. The derivative continuity can then be
+   * enforced by re-scaling shape functions so that the magnitude of each
+   * derivative is equivalent to the derivative on the reference interval.
+   */
+  virtual void
+  fill_fe_values(
+    const typename Triangulation<dim, spacedim>::cell_iterator &cell,
+    const CellSimilarity::Similarity                            cell_similarity,
+    const Quadrature<dim> &                                     quadrature,
+    const Mapping<dim, spacedim> &                              mapping,
+    const typename Mapping<dim, spacedim>::InternalDataBase &mapping_internal,
+    const dealii::internal::FEValuesImplementation::MappingRelatedData<dim,
+                                                                       spacedim>
+      &                                                            mapping_data,
+    const typename FiniteElement<dim, spacedim>::InternalDataBase &fe_internal,
+    dealii::internal::FEValuesImplementation::FiniteElementRelatedData<dim,
+                                                                       spacedim>
+      &output_data) const override;
+
+  using FiniteElement<dim, spacedim>::fill_fe_face_values;
```

Comment:
Why is this line needed?

Reply:
If I recall why this was added correctly, before this line was added the code automatically used the pre-existing function for Lagrange elements, and ignored the new version that we needed for correct results with Hermite elements.